src/histograms/charts/AbstractChart.ts

```diff
@@ -843,6 +855,58 @@ export abstract class AbstractChart extends AbstractHistogram {
     }
   }
 
+  /**
+   * When the selection is wider than the range of the data, adds "fake" buckets, for the x axis to be fully drawn
+   * @param data Data to plot
+   * @returns Extended data to fit the selection
+   */
+  protected extendData(data: Array<HistogramData>): Array<HistogramData> {
+    if (this.selectionInterval.startvalue === this.selectionInterval.endvalue) {
+      return data;
+    }
+
+    const bucketSize = this.getDataInterval(data);
+    let extendedData = new Array<HistogramData>();
+    if (+data[0].key > +this.selectionInterval.startvalue) {
+      let fakeDataKey = +data[0].key;
+      while (fakeDataKey > +this.selectionInterval.startvalue) {
+        fakeDataKey = fakeDataKey - bucketSize;
+        if (this.histogramParams.dataType === DataType.numeric) {
+          extendedData.push({
+            key: fakeDataKey,
+            value: 0
+          });
+        } else {
+          extendedData.push({
+            key: new Date(fakeDataKey),
+            value: 0
+          });
+        }
+      }
+    }
+
+    extendedData = extendedData.concat(data);
+    if (+data[data.length - 1].key < +this.selectionInterval.endvalue) {
+      let fakeDataKey = +data[data.length - 1].key;
+      while (fakeDataKey < +this.selectionInterval.endvalue) {
+        fakeDataKey += +bucketSize;
+        if (this.histogramParams.dataType === DataType.numeric) {
+          extendedData.push({
+            key: fakeDataKey,
+            value: 0
+          });
+        } else {
+          extendedData.push({
```

Comment:
I have the 'intuition' that maybe utc VS notUtc might be problematic when instantiating a new Date(xxx) ?

Reply:
After some tests, whether we use utc or not, the keys that the histogram have are still the same, the transformation happens when displaying the data ticks. The difference between buckets stay the same no matter if the histogram displays the data in utc or not. I am however going to add a safeguard regarding the number of buckets that can be plotted, as there is an intrinsic limit of 1000 if I am not mistaken.

Reply:
Ok great
For the limit, it' s 10000